setup/src/Magento/Setup/Module/Di/Code/Reader/FileScanner.php

```diff
@@ -264,7 +263,7 @@ protected function scan()
                         goto SCANNER_USE_CONTINUE;
                     }
 
-                    if ($this->tokenType == T_NS_SEPARATOR || $this->tokenType == T_STRING) {
+                    if (\in_array($this->tokenType, $namespaceContentTokenTypes, true)) {
```

Comment:
Could you replace it with a map and looking for a key in this map?
It works way faster, especially performance is critical for DI file scanner